### PR TITLE
snap, snappy: simplify extractKernelAssets and remove snap.File.Unpack

### PR DESCRIPTION
**Pull request by mvo5**
[based on https://github.com/ubuntu-core/snappy/pull/1123]

### CODE REVIEW

File: snap/squashfs/squashfs.go [hunk 1]
```diff
@@ -79,8 +82,7 @@ var runCommand = func(args ...string) error {
 	return nil
 }
 
-// Unpack unpacks the src (which may be a glob) into the given target dir.
-func (s *Snap) Unpack(src, dstDir string) error {
+func (s *Snap) unpack(src, dstDir string) error {
```
Comment (niemeyer) [inline snap/squashfs/squashfs.go]: Good choice, thanks!

Comment (niemeyer) [inline snap/squashfs/squashfs.go]: Should we get rid of the `src` parameter now?

File: snappy/kernel_os.go [hunk 2]
```diff
@@ -60,6 +55,13 @@ func removeKernelAssets(s snap.PlaceInfo, inter interacter) error {
 	return nil
 }
 
+func copyAll(src, dst string) error {
+	if output, err := exec.Command("cp", "-a", src, dst).CombinedOutput(); err != nil {
+		return fmt.Errorf("failed to copy %q -> %q: %s (%q)", src, dst, err, output)
```
Comment (niemeyer) [inline snappy/kernel_os.go]: s/failed to/cannot/

s/can not/cannot/ below as well, please.

Comment (niemeyer) [inline snappy/kernel_os.go]: Nevermind, sorry. This is from the base branch.

File: snappy/overlord.go [hunk 3]
```diff
@@ -116,7 +116,7 @@ func SetupSnap(snapFilePath string, sideInfo *snap.SideInfo, flags InstallFlags,
 	// FIXME: special handling is bad 'mkay
 	if s.Type == snap.TypeKernel {
 		if err := extractKernelAssets(s, snapf, flags, meter); err != nil {
-			return s, fmt.Errorf("failed to install kernel %s", err)
+			return s, fmt.Errorf("failed to install kernel: %s", err)
```
Comment (niemeyer) [inline snappy/overlord.go]: s/failed to/cannot/

File: snap/squashfs/squashfs_test.go [hunk 4]
```diff
@@ -91,23 +89,6 @@ func (s *SquashfsTestSuite) TestReadFile(c *C) {
 	c.Assert(string(content), Equals, "name: foo")
 }
 
-func (s *SquashfsTestSuite) TestUnpackGlob(c *C) {
```
Comment (niemeyer) [inline snap/squashfs/squashfs_test.go]: Should we keep this test and exercise it via Install with the env var set?

**Comment by niemeyer** [comment]
A few comments, and LGTM once you're happy about those.

**Comment by chipaca** [comment]
:+1:
should those `Cannots` be `cannots`?
Merging anyway.

**Pull request merged**